src/main/java/org/testng/TestRunner.java

```diff
@@ -728,18 +738,29 @@ private void privateRun(XmlTest xmlTest) {
         }
       } else {
         List<ITestNGMethod> freeNodes = graph.getFreeNodes();
+        if (true) {
+          Collections.sort(freeNodes);
+        }
 
         if (graph.getNodeCount() > 0 && freeNodes.isEmpty()) {
           throw new TestNGException("No free nodes found in:" + graph);
         }
 
         while (!freeNodes.isEmpty()) {
+          // This is sequential, let's try just running one at a time.
+          // TODO: To optimize this, we can know when we need to get more free nodes by knowing if 
```

Comment:
I don't have a great sense of how this change affects the performance of TestNG. If getting the free nodes is really expensive, then we'll want to look into optimizing this by not re-fetching the free nodes every time, but rather only when a method that other methods depend on is completed.